app/addons/documents/changes/actions.js

```diff
@@ -38,11 +45,59 @@ function (app, FauxtonAPI, ActionTypes) {
       });
     },
 
-    setChanges: function (options) {
+    initChanges: function (options) {
       FauxtonAPI.dispatch({
-        type: ActionTypes.SET_CHANGES,
+        type: ActionTypes.INIT_CHANGES,
         options: options
       });
+      this.getLatestChanges();
+    },
+
+    getLatestChanges: function () {
+      var params = {};
+
+      var lastSeqNum = changesStore.getLastSeqNum();
```

Comment:
Instead of default value for getLastSeqNum of null rather make it return `now`. Changes supports the `now` parameter. Then you don't need to do the if statement and instead just set the `since`, `feed` and `timeout`.